artiq/coredevice/comm_kernel.py

```diff
@@ -651,6 +655,26 @@ def read_exception_string():
             message = read_exception_string()
             params = [self._read_int64() for _ in range(3)]
 
+            # Add Channel name if any
+            m = re.match(r".+?(channel)\s{(\d+)}", message)
```

Comment:
For the 4th time: I don't want any string parsing/manipulation in there. The relevant exceptions are [known](https://github.com/m-labs/artiq/blob/master/artiq/coredevice/exceptions.py) and the channel numbers are already in the ``self.params``, already as integers.

Reply:
I think the problem is that the params is not consistent. channel number is located at `{0}` when it's `RTIOOverflow`, while located at `{1}` when it's `RTIOUnderflow`. And when it comes to `RTIODestinationUnreachable`, it could be both cases.
![Screenshot from 2022-07-05 15-13-03](https://user-images.githubusercontent.com/48083317/177270873-7697c18d-26d2-45d1-a546-585df10cf6ed.png)

Reply:
Sure